pkg/infoschema/issyncer/syncer.go

```diff
@@ -167,12 +168,11 @@ func (s *Syncer) refreshMDLCheckTableInfo(ctx context.Context) {
 		return
 	}
 	defer s.sysSessionPool.Put(se)
-	exec := sctx.GetRestrictedSQLExecutor()
 	domainSchemaVer := s.InfoSchema().SchemaMetaVersion()
 	// the job must stay inside tidb_ddl_job if we need to wait schema version for it.
 	sql := fmt.Sprintf(`select job_id, version, table_ids from mysql.tidb_mdl_info
 		where job_id >= %d and version <= %d`, s.minJobIDRefresher.GetCurrMinJobID(), domainSchemaVer)
-	rows, _, err := exec.ExecRestrictedSQL(ctx, nil, sql)
```

Comment:
`ExecRestrictedSQL` will create another session, it means we are using 2 internal sessions to run a single internal SQL, not sure why it do it

we need to change to the right side to make sure it works for cross keyspace scenario